webapp/packages/plugin-user-profile/src/UserMenu/UserMenu.tsx

```diff
@@ -27,7 +27,7 @@ export const UserMenu = observer(function UserMenu() {
     context.set(DATA_CONTEXT_USER, userInfoResource.data, id);
   });
 
-  if (!userInfoResource.isAuthenticated()) {
+  if (!userInfoResource.isData() || !userInfoResource.isAuthenticated) {
```

Comment:
seems like we can make `isAuthenticated` and `isAnonymous` type guards cuz they both guarantee that data is presented